src/Type/Definition/ResolveInfo.php

```diff
@@ -130,6 +131,12 @@ private function foldSelectionSet(SelectionSet $selectionSet, $descend)
                     $fragment = $this->fragments[$spreadName];
                     $fields += $this->foldSelectionSet($fragment->selectionSet, $descend);
                 }
+            } else if ($selectionAST instanceof InlineFragment) {
+                $fragmentFields = $this->foldSelectionSet($selectionAST->selectionSet, $descend - 1);
+                $fragmentType = $selectionAST->typeCondition->name->value;
+                foreach ($fragmentFields as $name => $subFields) {
+                    $fields[$fragmentType . ':' . $name] = $subFields;
```

Comment:
Maybe a new kind of nesting level instead of prefixing the key? Not very bc. Or returning a new, array accessible object with meta data like this? Could be bc, but might be overkill.

Also, I'm not sure this patch includes all cases: inline fragment, named fragment, inline fragment inside named fragment, etc. I've tried only 2 of many scenarios.